src/Stratis.Bitcoin.Features.Miner/PowMining.cs

```diff
@@ -175,7 +180,7 @@ public List<uint256> GenerateBlocks(ReserveScript reserveScript, ulong amountOfB
                     continue;
 
                 if (!this.ValidateAndConnectBlock(context))
-                    break;
+                    continue;
```

Comment:
So just to confirm if anything goes wrong with block connection we just continue mining yes?

Reply:
If a consensus rule fails it will throw, this flag is only true when the pre hash is not as the presented one which is ok.We are outside of the lock when we create the next block so in the mean time a new block could take the lead before we take the lock and make that check.